src/views/map.c

```diff
@@ -1795,15 +1795,15 @@ static gboolean _view_map_button_press_callback(GtkWidget *w, GdkEventButton *e,
   if(e->button == 1)
   {
     // check if the click was in a location form - crtl gives priority to images
-    if(lib->loc.main.id > 0 && !(e->state & GDK_CONTROL_MASK))
+    if(lib->loc.main.id > 0 && !dt_modifier_is(e->state, GDK_CONTROL_MASK))
     {
 
```

Comment:
This used to get executed if just SHIFT was pressed. From the surrounding code it looks like that was intentional?

Reply:
And it will still execute if Shift is the only modifier pressed (in fact, any combination except Control by itself).

Reply:
You're right of course; that's what I should  have said: the first bit now gets executed when CTRL+SHIFT is pressed whereas previously it wouldn't? But then, it doesn't seem to do anything useful if SHIFT is pressed (very possible that I misinterpret these function calls). So maybe the first condition should be
`if(lib->loc.main.id > 0 && dt_modifier_is(e->state, 0))`
and the later separate check for GDK_SHIFT_MASK can be removed?

(did I get that wrong again?)

Reply:
AIUI, that function wants to _skip_ parts if Control is pressed, and other parts if Shift is pressed.  I'd have to analyze the code in more detail to check whether both pieces should be skipped if both keys are pressed.  If it doesn't make sense to skip both, then the revised code is fine; if we want to be able to skip one or both, then additional checks are needed, or a `dt_modifier_includes` function that returns true if _at least_ the specified keys are held.

`dt_modifier_is(state, 0)` is true only if **no** modifiers are pressed, so any other checks guarded by the former will return false.

Reply:
On second thought, `dt_modifiers_include` is useful regardless, as it lets me replace a few other instances of checking for modifier bits and ensure that the future addition of Ctrl->Cmd  mapping is localized.  So I've gone ahead and added it.